README.md

```diff
@@ -49,3 +49,56 @@ tree.Sorter = func(u1, u2 fyne.URI) bool {
 <p align="center" markdown="1" style="max-width: 100%">
   <img src="img/widget-filetree.png" width="1024" height="880" alt="FileTree Widget" style="max-width: 100%" />
 </p>
+
+### CompletionEntry
+
+An extension of widget.Entry for displaying a popup menu for completion. The "up" and "down" keys on the keyboard are used to navigate through the menu, the "Enter" key is used to confirm the selection. The options can also be selected with the mouse. The "Escape" key closes the selection list.
+
+```go
+entry := widget.NewCompletionEntry([]string{})
+
+// When the use typed text, complete the list.
+entry.OnChanged = func(s string) {
+    // completion start for text length >= 3
+    if len(s) < 3 {
+        entry.HideCompletion()
+        return
+    }
+
+    // Make a search on wikipedia
+    resp, err := http.Get(
+        fmt.Sprintf(
+            "https://en.wikipedia.org/w/api.php?action=opensearch&search=%s", entry.Text,
+        ),
+    )
+    if err != nil {
+        entry.HideCompletion()
+    } else {
+        // Get the list of possible completion
+        results := make([][]interface{}, 0)
+        dec := json.NewDecoder(resp.Body)
+        dec.Decode(&results)
```

Comment:
We are not using `dec` anywhere else, I guess you could just do `json.NewDecoder(resp.Body).Decode(&results)` if you wanted to.